From 0771849495b4128cac2faf7d49c85c729fc48b20 Mon Sep 17 00:00:00 2001
From: Andy Gross <agross@codeaurora.org>
Date: Mon, 30 Jun 2014 21:18:39 -0500
Subject: [PATCH 182/182] qcom: Kconfig: Make drivers mutually exclusive

This patch makes sure QCOM ADM dmaengine and QCOM Nand cannot be enabled at the
same time.  This is an issue because the dma drivers will conflict with one
another.

Signed-off-by: Andy Gross <agross@codeaurora.org>
---
 drivers/dma/Kconfig      |    2 +-
 drivers/mtd/nand/Kconfig |    6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/dma/Kconfig
+++ b/drivers/dma/Kconfig
@@ -412,7 +412,7 @@ config QCOM_BAM_DMA
 
 config QCOM_ADM
 	tristate "Qualcomm ADM support"
-	depends on ARCH_QCOM || (COMPILE_TEST && OF && ARM)
+	depends on !MTD_QCOM_ADM && (ARCH_QCOM || (COMPILE_TEST && OF && ARM))
 	select DMA_ENGINE
 	select DMA_VIRTUAL_CHANNELS
 	---help---
--- a/drivers/mtd/nand/Kconfig
+++ b/drivers/mtd/nand/Kconfig
@@ -511,15 +511,15 @@ config MTD_NAND_XWAY
 	  to the External Bus Unit (EBU).
 
 config MTD_QCOM_DMA
-	tristate "QCMO NAND DMA Support"
-	depends on ARCH_QCOM && MTD_QCOM_NAND
+	tristate "QCOM NAND DMA Support"
+	depends on !QCOM_ADM && ARCH_QCOM && MTD_QCOM_NAND
 	default n
 	help
 	  DMA support for QCOM NAND
 
 config MTD_QCOM_NAND
 	tristate "QCOM NAND Device Support"
-	depends on MTD && ARCH_QCOM
+	depends on MTD && ARCH_QCOM && !QCOM_ADM
 	select CRC16
 	select BITREVERSE
 	select MTD_NAND_IDS
